### ISSUE TITLE
Setting GlobalStateManager.skipNameEntry will prevent MTM101BaldiDevAPI.OnSceneUnload from being executed.

### ISSUE THREAD

**Issue by Sakyce** [benjaminpants/MTM101BMDE#7]
As I understand from this source, the entry point you use most of the registering on your mod is in `NameManager.Awake`.
Although, the entry point should be somewhere around `MenuInitializer.Start` instead to support skipNameEntry.

I'll send further informations if I find something.

![image](https://github.com/benjaminpants/MTM101BMDE/assets/55030860/3d945d4e-9235-455e-9ea8-79af1280ce4a)

**Comment by Sakyce** [comment]
I'm bad at git but here is the changes suppoed to be done lol;
```cs
    //Handle patching appropiate functions to allow for the version number to be patched
    [HarmonyPatch(typeof(NameManager))]
    [HarmonyPatch("Awake")]
    public class InjectAPINameName
    {
        static void Postfix(NameManager __instance)
        {
            //the version number stuff
            Transform t = __instance.transform.parent.Find("Version Number");
            TMPro.TMP_Text text = t.gameObject.GetComponent<TMPro.TMP_Text>();
            text.text += "API " + MTM101BaldiDevAPI.VersionNumber;
            t.localPosition += new Vector3(0f, 28f);
        }
    }

    [HarmonyPatch(typeof(MenuInitializer))]
    [HarmonyPatch("Start")]
    public class ReloadScenes
    {
        static void Postfix()
        {
            // define all metadata before we call OnAllAssetsLoaded, so we can atleast be a bit more sure no other mods have activated and added their stuff yet.
            MTM101BaldiDevAPI.Instance.StartCoroutine(MTM101BaldiDevAPI.Instance.ReloadScenes());
        }
    }
   ```

**Comment by benjaminpants** [comment]
Fixed in 4.0, commit 10727e5
